packages/console/src/console-history/ConsoleHistory.tsx

```diff
@@ -13,43 +13,45 @@ interface ConsoleHistoryProps {
   language: string;
   openObject: (object: VariableDefinition) => void;
   disabled?: boolean;
+  supportsType(type: string): boolean;
+  iconForType(type: string): ReactElement;
 }
 
-class ConsoleHistory extends Component<
-  ConsoleHistoryProps,
-  Record<string, never>
-> {
-  static defaultProps = {
-    disabled: false,
-  };
-
-  static itemKey(i: number, item: ConsoleHistoryActionItem): string {
-    return `${i}.${item.command}.${item.result && item.result.message}.${
-      item.result && item.result.error
-    }`;
-  }
-
-  render(): ReactElement {
-    const { disabled, items, language, openObject } = this.props;
-    const historyElements = [];
-    for (let i = 0; i < items.length; i += 1) {
-      const item = items[i];
-      const historyElement = (
-        <ConsoleHistoryItem
-          key={ConsoleHistory.itemKey(i, item)}
-          disabled={disabled}
-          item={item}
-          openObject={openObject}
-          language={language}
-        />
-      );
-      historyElements.push(historyElement);
-    }
+function itemKey(i: number, item: ConsoleHistoryActionItem): string {
+  return `${i}.${item.command}.${item.result && item.result.message}.${
+    item.result && item.result.error
+  }`;
+}
 
-    return (
-      <div className="container-fluid console-history">{historyElements}</div>
+function ConsoleHistory(props: ConsoleHistoryProps): ReactElement {
+  const {
+    disabled = false,
+    items,
+    language,
+    openObject,
+    supportsType,
+    iconForType,
+  } = props;
+  const historyElements = [];
+  for (let i = 0; i < items.length; i += 1) {
+    const item = items[i];
+    const historyElement = (
+      <ConsoleHistoryItem
+        key={itemKey(i, item)}
+        disabled={disabled}
+        item={item}
+        openObject={openObject}
+        language={language}
+        supportsType={supportsType}
+        iconForType={iconForType}
+      />
     );
+    historyElements.push(historyElement);
   }
+
+  return (
+    <div className="container-fluid console-history">{historyElements}</div>
+  );
```

Comment:
Not that this was you, and no need to make this change, but now I'd prefer to write this with a map:
```
  return (
    <div className="container-fluid console-history">
      {items.map((item, i) => (
        <ConsoleHistoryItem
          key={itemKey(i, item)}
          disabled={disabled}
          item={item}
          openObject={openObject}
          language={language}
          supportsType={supportsType}
          iconForType={iconForType}
        />
      ))}
    </div>
  );
```